src/components/post/index.js

```diff
@@ -7,46 +7,63 @@ import Comment from "../comment";
 import EditPostModal from "../editPostModal";
 import ProfileCircle from "../profileCircle";
 import "./style.css";
+import { post, request, get } from "../../service/apiClient";
+
 
 const Post = ({
   name,
   date,
   content,
   comments = [],
-  likes = 0,
+  liked,
   id,
   setTriggerUpdate,
   currentUserName,
+  currentUserId
 }) => {
   const { openModal, setModal } = useModal();
 
   const userInitials = name.match(/\b(\w)/g);
   const [isLiked, setIsLiked] = useState(false);
-  const [like, setLike] = useState(likes);
+  const [like, setLike] = useState(liked.length);
 
-    const showModal = () => {
-        setModal('Edit post', 
-        <EditPostModal 
-            id={id} 
-            content={content} 
-            setTriggerUpdate={setTriggerUpdate}
-            name={name}
-            userInitials={userInitials}
-        />)
-        openModal()
-    }
 
-  const handleClick = () => {
-    setIsLiked((prevIsLiked) => !prevIsLiked);
-    setLike((prevLike) => (isLiked ? prevLike - 1 : prevLike + 1));
+  const showModal = () => {
+    setModal(
+      'Edit post',
+      <EditPostModal
+        id={id}
+        content={content}
+        setTriggerUpdate={setTriggerUpdate}
+        name={name}
+        userInitials={userInitials}
+      />
+    );
+    openModal();
   };
 
-  let likeIcon = null;
-  if (isLiked) {
-    likeIcon = <LikeRed />;
-  } else {
-    likeIcon = <Like />;
-  }
+  useEffect(() => {
+    for(let i = 0; i < liked.length; i++){
+      if (liked[i].userId === currentUserId){
+        setIsLiked(true);
+      }
+    }}, [currentUserId, liked])
+    
+  const handleClick = () => {
+    const requestData = { id };
+
+    if (isLiked) {
+      request("DELETE", `posts/${id}/like`, requestData).then(() => {
```

Comment:
rather than calling request directly, add a new method like the existing `get`/`post` for delete

Reply:
done!